rational/rational.go

```diff
@@ -0,0 +1,221 @@
+package rational
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+
+	wire "github.com/tendermint/go-wire"
+)
+
+var cdc *wire.Codec
+
+func init() {
+	cdc = wire.NewCodec()
+	cdc.RegisterInterface((*Rational)(nil), nil)
+	cdc.RegisterConcrete(Rat{}, "rat", nil)
+}
+
+// Rat - extend big.Rat
+type Rat struct {
+	*big.Rat `json:"rat"`
+}
+
+// Rational - big Rat with additional functionality
+type Rational interface {
+	GetRat() *big.Rat
+	Num() int64
+	Denom() int64
+	GT(Rational) bool
+	LT(Rational) bool
+	Equal(Rational) bool
+	IsZero() bool
+	Inv() Rational
+	Mul(Rational) Rational
+	Quo(Rational) Rational
+	Add(Rational) Rational
+	Sub(Rational) Rational
+	Round(int64) Rational
+	Evaluate() int64
+}
+
+var _ Rational = Rat{} // enforce at compile time
+
+// nolint - common values
+var (
+	Zero = Rat{big.NewRat(0, 1)}
+	One  = Rat{big.NewRat(1, 1)}
+)
+
+// New - create a new Rat from integers
+func New(Numerator int64, Denominator ...int64) Rat {
+	switch len(Denominator) {
+	case 0:
+		return Rat{big.NewRat(Numerator, 1)}
+	case 1:
+		return Rat{big.NewRat(Numerator, Denominator[0])}
+	default:
+		panic("improper use of New, can only have one denominator")
+	}
+}
+
+//NewFromDecimal - create a rational from decimal string or integer string
+func NewFromDecimal(decimalStr string) (f Rat, err error) {
+
+	// first extract any negative symbol
+	neg := false
+	if string(decimalStr[0]) == "-" {
+		neg = true
+		decimalStr = decimalStr[1:]
+	}
+
+	str := strings.Split(decimalStr, ".")
+
+	var numStr string
+	var denom int64 = 1
+	switch len(str) {
+	case 1:
+		if len(str[0]) == 0 {
+			return f, fmt.Errorf("not a decimal string")
```

Comment:
could've just used `errors.New`

Reply:
okay using `errors.New` I usually just use fmt.Errors to avoid confusing with other error packages that come up